packages/verkle/README.md

```diff
@@ -23,32 +23,79 @@ npm install @ethereumjs/verkle
 
 ## Usage
 
-### Initialization and Basic Usage
+### Initialization
+
+To initialize a verkle tree, we provide an async constructor `createVerkleTree` which returns a `VerkleTree` instance that is properly initialized with the required`VerkleCrypto` package that implements the necessary cryptographic primitives used by Verkle trees.
+
+```ts
+// ./examples/simple.ts#L7-L7
+
+const tree = await createVerkleTree()
+```
+
+Note, the current `VerkleCrypto` library we use internally is [`verkle-cryptography-wasm`](https://github.com/ethereumjs/verkle-cryptography-wasm) which is a WASM compilation of the [`rust-verkle`](https://github.com/crate-crypto/rust-verkle) library with some Javascript specific wrappers and helper methods.
+
+If you prefer to instantiate the verkle tree class directly, you can do so by passing in an already instantiated `VerkleCrypto` object and then initializing the root node manually.
 
 ```ts
+// ./examples/diyVerkle.ts
+
+import { MapDB, bytesToHex } from '@ethereumjs/util'
 import { VerkleTree } from '@ethereumjs/verkle'
-import { bytesToUtf8, utf8ToBytes } from '@ethereumjs/util'
+import { loadVerkleCrypto } from 'verkle-cryptography-wasm'
+
+const verkleCrypto = await loadVerkleCrypto()
+
+const main = async () => {
+  const tree = new VerkleTree({ verkleCrypto, db: new MapDB<Uint8Array, Uint8Array>() })
+  await tree['_createRootNode']()
+  console.log(bytesToHex(tree.root())) // 0x0000000000000000000000000000000000000000000000000000000000000000
+}
+
+void main()
+```
 
-const tree = new VerkleTree()
+### Getting and Putting Values
+
+Values are stored using a combination of a `stem` obtained through the `getVerkleStem` function exposed by `@ethereumjs/util`.
+
+Following the design goal of verkle trees of allowing efficient reads and writes of multiple values that are "close" to each other, the `get` and `put` methods take a stem as a first argument and then an array of "suffixes" (the 32nd byte) of the key used to access a value.
+
+#### Getting values
+
+When retrieving values given a stem `0x781f1e4238f9de8b4d0ede9932f5a4d08f15dae7` and suffixes `[0, 1]`, the `get` method would access the values stored at `0x781f1e4238f9de8b4d0ede9932f5a4d08f15dae70000` and `0x781f1e4238f9de8b4d0ede9932f5a4d08f15dae70001`.
```

Comment:
I might be missing something here, but it concats `0000` or `0001` to a 20-byte hex string? (Not something like 31 or 32 bytes which seems to be  expected here?)

Reply:
Ah ok, it seems that this is an address. Then the docs should be updated that the suffix is thus added to the stem (concatenated). A stem thus seems to be able to have variable sizes (?)

Reply:
Oops, I must have copied the address instead of the stem I generated for it here.  Let me generate a valid stem and will update this.